### PR TITLE
Add hosted mode support to certificate controller

### PR DESCRIPTION
**Pull request by gparvin**
We need hosted mode support for the certificate
policy controller.  This adds it and some tests
to make sure it keeps working.

Refs:
 - https://github.com/stolostron/backlog/issues/26177

Signed-off-by: Gus Parvin <gparvin@redhat.com>

### CODE REVIEW

File: test/e2e/e2e_suite_test.go [hunk 1]
```diff
@@ -41,7 +41,7 @@ var (
 
 func TestE2e(t *testing.T) {
 	RegisterFailHandler(Fail)
-	RunSpecs(t, "Config policy controller e2e Suite")
+	RunSpecs(t, "Certificate policy controller e2e Suite")
```
Comment (ChunxiAlexLuo) [inline test/e2e/e2e_suite_test.go:44]: Nice catch

File: test/utils/utils.go [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+// Pause sleep for given seconds
+func Pause(s uint) {
+	if s < 1 {
+		s = 1
+	}
+
+	time.Sleep(time.Duration(float64(s)) * time.Second)
+}
+
+// ParseYaml read given yaml file and unmarshal it to &unstructured.Unstructured{}
```
Comment (ChunxiAlexLuo) [inline test/utils/utils.go:35]: Same

File: test/utils/utils.go [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+)
+
+// Pause sleep for given seconds
```
Comment (ChunxiAlexLuo) [inline test/utils/utils.go:26]: Is this function used somewhere else？

File: test/utils/utils.go [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+	Expect(err).To(BeNil())
+
+	yamlPlc := &unstructured.Unstructured{}
+	err = yaml.Unmarshal(yamlFile, yamlPlc)
+	Expect(err).To(BeNil())
+
+	return yamlPlc
+}
+
+// GetClusterLevelWithTimeout keeps polling to get the object for timeout seconds until wantFound is met
```
Comment (ChunxiAlexLuo) [inline test/utils/utils.go:47]: Same

**Review by ckandag — APPROVED** [review]

File: test/utils/utils.go (hunk 4 above)
Comment (gparvin) [reply test/utils/utils.go:47]: Nope, thanks -- will remove

**Comment by gparvin** [comment]
Addressed review comments.  Thanks Alex!

**Review by ChunxiAlexLuo — APPROVED** [review]
/lgtm

**Review by ckandag — APPROVED** [review]

**Pull request merged**